### PR TITLE
Update HOHQMeshMain.f90

### PR DESCRIPTION
**Pull request by DavidAKopriva**
Add help to be printed with the -help option.

### CODE REVIEW

File: Source/HOHQMeshMain.f90 [hunk 1]
```diff
@@ -162,13 +161,13 @@ SUBROUTINE ReadCommandLineArguments(version, test, generateTest, controlFileName
 
 
          IF ( CommandLineArgumentIsPresent("-version") )     THEN
-            PRINT *, "HOMesh Version ", version
+            PRINT *, "HOHQMesh Version ", version
             STOP
          END IF
 
          IF ( CommandLineArgumentIsPresent("-help") )     THEN
```
Comment (sloede) [inline Source/HOHQMeshMain.f90:168]:
```suggestion
         IF ( CommandLineArgumentIsPresent("-help") .OR. CommandLineArgumentIsPresent("--help") )     THEN
```
I know that the HOHQMesh standard is to eschew command line arguments with two dashes, but OTOH, `--help` is _the_ go-to command I usually run first for an unknown program, thus maybe it wouldn't hurt to allow it just for the help?

Comment (sloede) [reply Source/HOHQMeshMain.f90:168]: Actually, I wouldn't modify it 😅: I'd say `-help` is the correct command to use, and `--help` is just added for sanity for users who do not bother to read the docs but like to explore things interactively (such as myself 😬).

But I have no strong opinion on this, thus I'm good either way.

Comment (andrewwinters5000) [reply Source/HOHQMeshMain.f90:168]: Okay, I don't really mind either way. I will resolve my conversation below.

**Comment by DavidAKopriva** [comment]
OK, I've added three options to get help: -help, --help and -h. The help message gets printed if there is an error.

**Review by sloede — COMMENTED** [review]
With the one redundancy removed, this LGTM!

File: Source/HOHQMeshMain.f90 [hunk 2]
```diff
@@ -147,57 +147,116 @@ END PROGRAM HOQMeshMain
...
-            WRITE(stderr,*)  "No help available yet. Sorry!"
-            ERROR STOP "No help available"
+            CALL PrintHelpMessage(OUTPUT_UNIT)
+            STOP
+         END IF
+
+         IF ( CommandLineArgumentIsPresent("-h") )     THEN
+            CALL PrintHelpMessage(OUTPUT_UNIT)
+            STOP
+         END IF
```
[the remark this answers is not in the document]
Comment (DavidAKopriva) [reply Source/HOHQMeshMain.f90:179]: That's true. I'll remove.

**Comment by DavidAKopriva** [comment]
I still need to put in some test coverage. Will do that next.

**Comment by DavidAKopriva** [comment]
It passes codecov/project but not codecov/patch. Not sure what that means, and I'm not sure how to test command line arguments when those are passed through from the user. Unless there is some way to spoof the CLAs.

File: Source/HOHQMeshMain.f90 [hunk 3]
```diff
@@ -201,3 +200,33 @@ SUBROUTINE ReadCommandLineArguments(version, test, generateTest, controlFileName
...
+!//////////////////////////////////////////////////////////////////////// 
+! 
+      SUBROUTINE PrintHelpMessage()
+         USE, INTRINSIC :: iso_fortran_env, only : OUTPUT_UNIT
+         IMPLICIT NONE
+         WRITE(OUTPUT_UNIT,*) "HOHQMesh Help..."
+         WRITE(OUTPUT_UNIT,*) "Invocation:"
+         WRITE(OUTPUT_UNIT,*) "	./HOHQMesh [options]"
+         WRITE(OUTPUT_UNIT,*) "Options:"
+         WRITE(OUTPUT_UNIT,*) "-help"
```
[the remark this answers is not in the document]
Comment (fhindenlang) [reply Source/HOHQMeshMain.f90]: well the license says that you can copy it, so I just copied it into my code, its not something that need maintenance. But no worries, its your choice ;)

**Comment by sloede** [comment]
I improved coverage in [cddf7f7](https://github.com/trixi-framework/HOHQMesh/pull/121/commits/cddf7f7767cd95552ad918c7479e82b87cf45f41). LGTM now!

**Review by sloede — APPROVED** [review]

**Review by andrewwinters5000 — APPROVED** [review]
LGTM!

**Pull request merged**